### DISCUSSION TITLE
New CIDRAM fav icon?

### DISCUSSION THREAD

**Discussion by 737simpilot** [CIDRAM/CIDRAM#482] [Q&A] [answered]
I recently noticed that CIDRAM is loading a different fav icon than I'm used to. Is this new for CIDRAM or something odd on my end? I can hardly see it, but it looks like a toy soldier or something.

**Comment by Maikuolan** [comment, marked as answer]
I did recently update the CIDRAM favicon, so something new for CIDRAM. The old favicon only supported 16x16 dimensions, but I've added 5 different dimension variants for the newly updated favicon, to hopefully have it look slightly better at different screen resolutions and for different favicon contexts (e.g., the favicon dimensions preferred by browsers when displaying the list of bookmarked/favourited links, versus displaying the "quick links" at the browser's start page, versus the favicon dimensions preferred by the Windows tiles system for webapps, versus that preferred by various Apple devices and so on).

I'm not going to bother covering every possible preferred dimension, because there are too many, and I don't want us ending up with a severely bloated favicon file if it can be helped. But I figure, providing just 5 of the more common preferences should be reasonably helpful without bloating it too much. Also, the 5 different dimension variants are only served when accessing the front-end, whereas the standard access denied page seen by blocked requests is served only a much smaller version of the favicon which still only supports 16x16 dimensions, the same as the old favicon (the newly updated favicon, as it stands right now, is 11.4 KB when served at the front-end, or 318 bytes when served at the access denied page, both of which are still a hell of a lot more than the 124 bytes that the old favicon was, but also isn't too unreasonable given the average size of the average favicon at most websites nowadays, typical load size of pages at modern websites and so on).

![favicon](https://github.com/CIDRAM/CIDRAM/assets/12571108/a3c6b9ef-5bb1-4695-8151-b76f7392fe52)

The "toy soldier" in the newly updated favicon is a scaled-down copy of the logo created by @gizmecano at #122 (it's a great image, and I've felt that it would be great to be able to start actually using it somewhere since the time I first saw it, but I wasn't really ever sure *where* exactly it should be used; but then, when redoing the favicon, figured I may as well experiment and see whether it still works when scaled down for some of the larger favicon variants, as that could serve as a starting point, if it worked). There's a limit, of course, to how much an image can be scaled down before it starts losing all its finer details, gradually becomes unrecognisable, and eventually, completely incoherent. There's enough finer detail still evident in the 64x64 and 48x48 variants that I figured it would be worth including it in, but with the remaining smaller variants, there's barely enough pixels for a few letters to be recognisable, so I kept those as simple as possible.

The newly updated favicon mightn't be super impressive, but it's still an improvement over the old favicon, I think.

When you say you can "hardly see it" though, I'm curious: In what context are you seeing it at all? From my own experience, browsers should typically display the 16x16 variant for bookmarks/favourites, or the 32x32 variant for "quick links". Any desktop shortcuts pointing to the the CIDRAM front-end might use the larger variants, and "tiles", or any other systems which typically prefer huge images, would most likely default to the 64x64 variant when they find that their own preferred dimensions aren't supported, and that the 64x64 variant is the next largest/closest one available. So, you'll likely see the toy soldier appear in the favicon at "tile" systems and such. But, I wouldn't normally expect it to appear for bookmarks/favourites or quick links, given that it isn't present in those smaller variants. If it does appear, I would question whether the browser is perhaps selecting the wrong variant and then shrinking it down, instead of selecting the variant already provided for the desired dimensions.

**Comment by 737simpilot** [comment]
I'm using the browser Pale Moon and it is in there where the favicon shows up. I don't even think Firefox (in their infinite wisdom) shows favicons anymore.

**Comment by 737simpilot** [comment]
![img](https://i.postimg.cc/SNKKzLHj/fcvxdbdf.jpg)


In Firefox it does show up but in the tab. It's the same size.

**Comment by Maikuolan** [comment]
Looks like it's showing the 64x64 variant there. Super weird. o.0

Good to know though, at least. For comparison, here's how Edge handles the new favicon (might want to open the image in a new tab to view it properly; the discussions page squishes it a bit when viewing it here):

![Edge](https://github.com/CIDRAM/CIDRAM/assets/12571108/0e984949-10a2-4dd2-a7f8-9ce08cc64844)

I wasn't aware that not all browsers will automatically select the best dimensions for a favicon when multiple dimensions are available. I'll need to investigate that further, I think. I'm curious whether that's a bug, an oversight, or something intentional.

**Comment by 737simpilot** [comment]
It's interesting because I always saw the 16x16 variant before. I did recently update Pale Moon so maybe that's it.

**Reply by 737simpilot** [reply]
On the other hand, Firefox shows the same favicon.

**Comment by 737simpilot** [comment]
So I guess we have a logo? Can the organization logo be updated?

**Reply by Maikuolan** [reply]
Ideally, if possible, I'm aiming to have something around 400x400 or 420x420, so that it'll work well with how GitHub scales profile images and such across the platform. Not sure how the favicon variants would look under that same scaling. I'll look into that soon, and failing all else, I can give the 64x64 favicon variant a try for the organisation image. :-)

**Comment by Maikuolan** [comment]
Wondering whether I should nix some of the larger variants from the favicon entirely. If not all browsers are correctly selecting the most appropriate variant, it might be better to not include the larger ones at all. :-/

On a related note.. How do you feel about this for an organisation logo?

![Logo](https://github.com/CIDRAM/CIDRAM/assets/12571108/ae866e6f-0988-4ef3-b3f4-c7666eba0cd5)

Not the most exciting image, but all the favicons looked super ugly when scaled up to the sizes recommended for organisation logos. Conversely, adding in too much detail into an organisation logo, when the logo gets scaled down for displaying at the profiles of organisation members, most of that detail is lost, and it becomes difficult to really distinguish anything in the image.

Going the "pixel art" approach, and just keeping it simple like that, the level of apparent detail looks the same regardless of whether scaled up or scaled down (so there shouldn't be any loss of detail when viewing it at profiles), the space is filled up enough as so that it doesn't look too "empty" (even though what's filling it up is just pixel art-style text, but.. eh.. I'm not a graphic artist or graphics designer, so if I can utilise some cheap tricks to get it done, so be it, I guess).

Take in mind, we can always change it again if we come up with something better at a later date; it doesn't need to be permanent. But, good to have something at least a little better than the default organisation placeholder image.

**Comment by Maikuolan** [comment]
Hrmf.. Spoke too soon about loss of detail. Still seems legible at 64x64, but becomes illegible at 48x48, 32x32, 16x16, etc. :-/

Well.. I've gone ahead and changed it for now anyway, so that we're not using the default placeholder image from GitHub, at least. We can work on something better whenever, I guess.

**Comment by Maikuolan** [comment]
New attempt:

![Logo2](https://github.com/CIDRAM/CIDRAM/assets/12571108/8f603427-f09d-493d-a1fc-e52a95e9d929)

**Comment by 737simpilot** [comment]
That's not bad I guess. Better than what we had before.
